src/tribler-core/tribler_core/modules/tests/bandwidth_accounting/test_bandwidth_endpoint.py

```diff
@@ -1,45 +1,59 @@
-from ipv8.test.mocking.ipv8 import MockIPv8
-
 import pytest
 
+from ipv8.test.mocking.ipv8 import MockIPv8
 from tribler_core.modules.bandwidth_accounting import EMPTY_SIGNATURE
+from tribler_core.modules.bandwidth_accounting.bandwidth_endpoint import BandwidthEndpoint
 from tribler_core.modules.bandwidth_accounting.community import BandwidthAccountingCommunity
+from tribler_core.modules.bandwidth_accounting.database import BandwidthDatabase
 from tribler_core.modules.bandwidth_accounting.settings import BandwidthAccountingSettings
 from tribler_core.modules.bandwidth_accounting.transaction import BandwidthTransactionData
-from tribler_core.restapi.base_api_test import do_request
+from tribler_core.restapi.base_api_test import do_app_request
 from tribler_core.utilities.unicode import hexlify
 
 
 @pytest.fixture
-async def mock_ipv8(session):
-    db_path = session.config.state_dir / "bandwidth.db"
-    ipv8 = MockIPv8("low", BandwidthAccountingCommunity, database_path=db_path,
+def bandwidth_database(tmp_path, peer_key):
+    return BandwidthDatabase(db_path=tmp_path / "bandwidth.db", my_pub_key=b"0000")
+
+
+@pytest.fixture
+async def bw_community(tmp_path, bandwidth_database):
+    ipv8 = MockIPv8("low", BandwidthAccountingCommunity,
+                    database=bandwidth_database,
                     settings=BandwidthAccountingSettings())
-    session.bandwidth_community = ipv8.overlay
-    yield ipv8
+    community = ipv8.get_overlay(BandwidthAccountingCommunity)
+    # Dumb workaround for MockIPv8 not supporting key injection
```

Comment:
Noted. I created a PR for this in IPv8: https://github.com/Tribler/py-ipv8/pull/1016